File: prob-1.java
//Problem-1

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
//TC:o(n)
//sc:o(maxdepth)


class Solution {
    public int kthSmallest(TreeNode root, int k) 
    {
       Stack<TreeNode> stack = new Stack<>();
       while(root!=null || !(stack.isEmpty()) ){
           while(root!=null)
           {
              stack.push(root);
              root=root.left;    
           }
           TreeNode p = null;
           if(root==null && !(stack.isEmpty()) )
           {  
             p=stack.pop();
             k--;
             if(k==0) 
                 return p.val;
           }
           root = p.right;
       }  
        return -1;
    }
}

File: prob1.py
# Time Complexity :O(n)
# Space Complexity :O(h)
# Leet Code: Yes

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.count = k
        self.result = TreeNode()
        self.helper(root)
        return self.result.val
        

    def helper(self, root):
            # base
            if root is None:
                return None

            # logic
            self.helper(root.left)
            self.count -= 1
            if self.count == 0:
                self.result = root
            self.helper(root.right)
